src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -0,0 +1,374 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.VisualStudio.TestPlatform.Utilities;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.Intrinsics.Arm;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Xunit;
+using Xunit.Abstractions;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace Microsoft.DotNet.Installer.Tests;
+
+public class LinuxInstallerTests : IDisposable
+{
+    private readonly DockerHelper _dockerHelper;
+    private readonly string _tmpDir;
+    private readonly string _contextDir;
+
+    private readonly string[] RpmDistroImages =
+    [
+        "mcr.microsoft.com/dotnet/runtime-deps:9.0-azurelinux3.0"
+    ];
+
+    private readonly string[] DebDistroImages =
+    [
+        "mcr.microsoft.com/dotnet/runtime-deps:9.0-bookworm-slim"
+    ];
+
+    private const string NetStandard21RpmPackage = @"https://dotnetcli.blob.core.windows.net/dotnet/Runtime/3.1.0/netstandard-targeting-pack-2.1.0-x64.rpm";
+    private const string NetStandard21DebPackage = @"https://dotnetcli.blob.core.windows.net/dotnet/Runtime/3.1.0/netstandard-targeting-pack-2.1.0-x64.deb";
+
+    private enum DistroType
+    {
+        Rpm,
+        Deb
+    }
+
+    private ITestOutputHelper OutputHelper { get; set; }
+
+    public LinuxInstallerTests(ITestOutputHelper outputHelper)
+    {
+        OutputHelper = outputHelper;
+        _dockerHelper = new DockerHelper(OutputHelper);
+
+        _tmpDir = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        Directory.CreateDirectory(_tmpDir);
+        _contextDir = Path.Combine(_tmpDir, Path.GetRandomFileName());
+        Directory.CreateDirectory(_contextDir);
+
+        InitializeContext();
+    }
+
+    public void Dispose()
+    {
+        try
+        {
+            Directory.Delete(_tmpDir, recursive: true);
+        }
+        catch
+        {
+        }
+    }
+
+    [Fact]
+    public void RunScenarioTestsForAllDistros()
+    {
+        if (Config.TestRpmPackages)
```

Comment:
Ideally these would be separate testcases.  One of the primary reasons for this is that if one fails you would still get a read on the other.  Also having explicitly testcases helps make it more obvious what is being tested.

Reply:
Fixed with https://github.com/dotnet/sdk/commit/64fa1b8e6d800687ab49eae0b2a483e937a83e73